### PR TITLE
Fetch: Response's trailer

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/fetch/issues/473 for context.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 1f938ee13cef917dbd5bf4154365d1a7f90d99c2
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4811/fetch/api/response/response-idl.html">/fetch/api/response/response-idl.html</a></summary>


|                                                            Subtest                                                             | Results |
|--------------------------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                                                | OK      |
| `Response interface: new Response() must inherit property "status" with the proper type (4)`                                   | PASS    |
| `Response interface: new Response() must inherit property "statusText" with the proper type (6)`                               | PASS    |
| `Response interface: operation error()`                                                                                        | PASS    |
| `Response interface: operation arrayBuffer()`                                                                                  | PASS    |
| `Response interface: new Response() must inherit property "redirect" with the proper type (1)`                                 | PASS    |
| `Response interface: new Response() must inherit property "formData" with the proper type (14)`                                | FAIL    |
| `Response interface: attribute url`                                                                                            | PASS    |
| `Response interface: new Response() must inherit property "headers" with the proper type (7)`                                  | PASS    |
| `Response interface: attribute trailer`                                                                                        | FAIL    |
| `Response interface: new Response() must inherit property "arrayBuffer" with the proper type (12)`                             | PASS    |
| `Response interface: attribute type`                                                                                           | PASS    |
| `Response interface: new Response() must inherit property "bodyUsed" with the proper type (11)`                                | PASS    |
| `Response interface: existence and properties of interface object`                                                             | PASS    |
| `Response interface: attribute headers`                                                                                        | PASS    |
| `Response interface: attribute body`                                                                                           | PASS    |
| `Response interface: operation redirect(USVString,unsigned short)`                                                             | PASS    |
| `Response interface: operation formData()`                                                                                     | FAIL    |
| `Response interface: new Response() must inherit property "trailer" with the proper type (8)`                                  | FAIL    |
| `Response interface: attribute status`                                                                                         | PASS    |
| `Response interface: new Response() must inherit property "ok" with the proper type (5)`                                       | PASS    |
| `Response interface: attribute bodyUsed`                                                                                       | PASS    |
| `Response interface: new Response() must inherit property "text" with the proper type (16)`                                    | PASS    |
| `Response interface: operation text()`                                                                                         | PASS    |
| `Response interface object name`                                                                                               | PASS    |
| `Response interface: existence and properties of interface prototype object`                                                   | FAIL    |
| `Response interface: new Response() must inherit property "clone" with the proper type (9)`                                    | PASS    |
| `Response interface object length`                                                                                             | PASS    |
| `Response must be primary interface of new Response()`                                                                         | PASS    |
| `Response interface: attribute statusText`                                                                                     | PASS    |
| `Response interface: new Response() must inherit property "error" with the proper type (0)`                                    | PASS    |
| `Response interface: new Response() must inherit property "body" with the proper type (10)`                                    | PASS    |
| `Response interface: new Response() must inherit property "blob" with the proper type (13)`                                    | PASS    |
| `Response interface: new Response() must inherit property "url" with the proper type (3)`                                      | PASS    |
| `Response interface: calling redirect(USVString,unsigned short) on new Response() with too few arguments must throw TypeError` | PASS    |
| `Response interface: new Response() must inherit property "json" with the proper type (15)`                                    | PASS    |
| `Response interface: operation blob()`                                                                                         | PASS    |
| `Response interface: existence and properties of interface prototype object's "constructor" property`                          | PASS    |
| `Response interface: operation clone()`                                                                                        | PASS    |
| `Response interface: attribute ok`                                                                                             | PASS    |
| `Response interface: new Response() must inherit property "type" with the proper type (2)`                                     | PASS    |
| `Stringification of new Response()`                                                                                            | PASS    |
| `Response interface: operation json()`                                                                                         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4811/fetch/api/response/response-trailer.html">/fetch/api/response/response-trailer.html</a></summary>


|     Subtest      | Results |
|------------------|---------|
|                  | OK      |
| `trailer() test` | FAIL    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 1f938ee13cef917dbd5bf4154365d1a7f90d99c2
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations

File: fetch/api/response/response-trailer.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    assert_equals(res.headers.get("Trailer"), "X-Test-Me")
+    assert_equals(res.headers.get("X-Test-Me"), null)
+    assert_equals(res.headers.get("Content-Type"), "text/plain")
+    return Promise.all([
+      res.text().then(text => {
+        assert_equals(text, "First chunk\r\nSecond chunk\r\nYet another (third) chunk\r\nYet another (fourth) chunk\r\n")
+      }),
+      res.trailer.then(trailers => {
+        assert_equals(trailers.get("X-Test-Me"), "Trailer header value")
+        assert_throws(new TypeError, trailers.append("Immutable-My-Ass", "Hi!"))
```
Comment (domenic) [inline fetch/api/response/response-trailer.html]: The second arg to `assert_throws` must be a function, but trailers.append does not return a function

Comment (annevk) [inline fetch/api/response/response-trailer.html]: Thanks! It's hard to test when there's no implementations.

**Comment by domenic** [comment]
I mostly just did a drive-by review on syntax; I don't feel qualified to review anything around the actual semantics of trailers.

**Comment by annevk** [comment]
Since this is a new feature and implementations will have to implement anyway, syntax review seems good enough.

**Pull request merged**